### PR TITLE
Printing logs at the end

### PR DESCRIPTION
**Pull request by Tulsishah**
### Description
Our test logs are currently unorganized, with output from different packages intermixed. This makes it extremely difficult to troubleshoot failures because we can't easily isolate logs from specific tests.

Create a temporary file to store log file names from a background process. Create an array from it and print logs from the given log files, then delete them.

### Link to the issue in case of a bug fix.
NA

### Testing details
1. Manual - NA
2. Unit tests - NA
3. Integration tests - Automated

### CODE REVIEW

File: perfmetrics/scripts/run_e2e_tests.sh [hunk 1]
```diff
@@ -145,6 +156,22 @@ function run_parallel_tests() {
...
+  rm "$TEST_LOGS_FILE"
+  for test_log_file in "${TEST_LOGS_ARRAY[@]}"
+  do
+    log_file=${test_log_file}
+    if [ -f "$log_file" ]; then
+      echo "=== Log for ${test_log_file} ==="
+      cat "$log_file"
+      echo "========================================="
+      # Remove the log file
+      rm "$log_file"
```
Comment (sethiay) [inline perfmetrics/scripts/run_e2e_tests.sh:170]: nit: there is no harm in not removing the log file. incase someone ssh into the VM, they will be able to see the log files.

File: perfmetrics/scripts/run_e2e_tests.sh [hunk 2]
```diff
@@ -145,6 +156,22 @@ function run_parallel_tests() {
   return $exit_code
 }
 
+function print_test_logs() {
+  readarray -t TEST_LOGS_ARRAY < "$TEST_LOGS_FILE"
```
Comment (sethiay) [inline perfmetrics/scripts/run_e2e_tests.sh:160]: nit: TEST_LOGS_ARRAY should be in small as it is a local variable ?

File: perfmetrics/scripts/run_e2e_tests.sh (hunk 1 above)
Comment (Tulsishah) [reply perfmetrics/scripts/run_e2e_tests.sh:170]: Okay, If we want to keep it then I changed path to `/tmp` for not creating redundant files in current directory.

Comment (sethiay) [reply perfmetrics/scripts/run_e2e_tests.sh:170]: please remove `rm` then ?

Comment (Tulsishah) [reply perfmetrics/scripts/run_e2e_tests.sh]: Sorry missed. Done.

**Review by sethiay — APPROVED** [review]

**Pull request merged**